### PR TITLE
IBX-461: Fixed roles with deleted limitations

### PR DESCRIPTION
**Pull request by barw4**
| Question      | Answer
| ------------- | ---
| Tickets       | [IBX-461](https://jira.ez.no/browse/IBX-461)
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Follow up PR of https://github.com/ezsystems/repository-forms/pull/347.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Review by adamwojs — APPROVED** [review]

**Review by Nattfarinn — APPROVED** [review]

File: src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php [hunk 1]
```diff
@@ -84,29 +84,28 @@ public function dataProviderForReverseTransform(): array
...
-        return [
-            [NotFoundException::class],
-            [UnauthorizedException::class],
-        ];
+        $this->expectException(TransformationFailedException::class);
+
+        $this->locationService
+            ->method('loadLocation')
+            ->willThrowException(
+                $this->createMock(UnauthorizedException::class)
```
Comment (Steveb-p) [inline src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php:105]: Inconsistent static / `$this` usage. Keep one or the other I guess?

I mean, if you want to make Andrew extra happy :heart_eyes: then use static variant :stuck_out_tongue_winking_eye:

**Review by Steveb-p — APPROVED** [review]
I'm ok with the general solution.

File: src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php (hunk 1 above)
Comment (alongosz) [reply src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php:105]: > Inconsistent static / `$this` usage. Keep one or the other I guess?
> 
> I mean, if you want to make Andrew extra happy  then use static variant

AFAICS createMock is not static :stuck_out_tongue_winking_eye:

**Review by alongosz — APPROVED** [review]

File: src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php (hunk 1 above)
Comment (Steveb-p) [reply src/lib/Tests/Form/DataTransformer/UDWBasedValueViewTransformerTest.php:105]: Darn, you got me.

_And I would get away with it, were it not for those meddling kids..._


Disregard :smile:

**Review by tomaszszopinski — APPROVED** [review]
QA approved on Ibexa DXP 3.3.4 & 3.3.4-dev with diff & patch

**Comment by lserwatka** [comment]
You can merge it up

**Comment by barw4** [comment]
Merged into `master`: https://github.com/ezsystems/ezplatform-admin-ui/commit/c245fe113f78d2c2102425ae4d6f5bf19130e0db

**Pull request merged**